PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageOne.java

```diff
@@ -15,44 +22,61 @@
 import org.scalaide.ui.ScalaImages;
 import org.scalaide.ui.internal.wizards.NewScalaProjectWizardPageOne;
 
+import com.microsoft.azure.hdinsight.projects.SparkVersion;
 import com.microsoft.azuretools.adauth.StringUtils;
 import com.microsoft.tooling.msservices.components.DefaultLoader;
 
 class HDInsightScalaPageOne extends NewScalaProjectWizardPageOne {
 	
 	private SparkLibraryOptionsPanel sparkLibraryOptionsPanel;
+	private HDInsightsScalaProjectWizard parent = null;
 	
 	public HDInsightScalaPageOne() {
 		super();
+
+	}
+	
+	public void setParent(HDInsightsScalaProjectWizard parent) {
+		this.parent = parent;
 	}
 	
 	@Override
 	public boolean canFlipToNextPage() {
-		final String jarPathString = sparkLibraryOptionsPanel.getSparkLibraryPath();
-		if(StringUtils.isNullOrEmpty(jarPathString)) {
-			return false;
+		if (!sparkLibraryOptionsPanel.useMaven) {
```

Comment:
Should not. This is just for safety check. In case it is called before sparkLibraryOptionsPanel is initialized.

---
In reply to: [136772513](https://github.com/Microsoft/azure-tools-for-java/pull/984#discussion_r136772513) [](ancestors = 136772513)